% Document information:
\documentclass[10pt]{article}
\author{Brian Seitz}
\title{Bioinformatics Final Project Proposal}
\date{\today\vspace{2em}}

% Attach packages:
\usepackage{titlesec}
\usepackage{titling}
\usepackage{fontspec}
\usepackage[margin=0.75in]{geometry}
\usepackage{polyglossia}
\usepackage[autostyle, english=american]{csquotes}
\usepackage[backend=bibtex, style=authoryear, citestyle=authoryear]{biblatex}

% Specify bibliography file:
\addbibresource{bioinformatics_final.bib}

% Specify fonts and formatting:
\newfontfamily\headingfont[]{Helvetica Neue}
\titleformat*{\section}{\Large\headingfont}
\titleformat*{\subsection}{\large\headingfont}
\titleformat*{\subsubsection}{\large\headingfont}
\renewcommand{\maketitlehooka}{\headingfont}
\setmainfont[]{Georgia}

\setlength\fboxsep{3mm}

\setmainlanguage{english}

\begin{document}
\maketitle

% First set global chunk options for Knitr:
<<chunk_settings, eval = TRUE, echo = FALSE>>=
opts_chunk$set(eval = TRUE, echo = TRUE, message = FALSE, warning = FALSE,
               fig.width = 7.5, fig.align = "center")
@

<<Load_Packages, echo = FALSE>>=
### Load Required Packages:
### For nicer manipulation of data.frames:
library(tidyr)
library(dplyr)
library(ggplot2)

### Packages with data:
# library(ALL)
# data(ALL)
# library("annaffy")
# library("hgu95av2.db")
# library(multtest)
# data(golub)

### Packages with stat tests:
# library(lmtest)
# library(nortest)
library(genefilter)
# library(limma)
# library(pvclust)

### Packages for going fast: 
library(parallel)
library(foreach)
library(doMC)

# LaTeX sanitizing function (to deal with underscores, other problems)
latexSanitize <- function (x) {
  output <- gsub("_", "\\\\_", x)
  output <- gsub("&", "\\\\&", output)
  output <- gsub("%", "\\\\%", output)
  return(output)
}

# Function to trim leading Xs inserted into column names: 
trim_x <- function(x) {
  if(grepl("^X", x)) {
  return(substr(x, 2, nchar(x)))
  } else return(x)
}

# How many cores to use?
num_cores <- detectCores()
if(num_cores >= 2) {num_cores <- num_cores - 1}

# Color palatte for graphs:
colorPalette <- c("#0072B2", "#E69F00", "#56B4E9", "#009E73", 
                  "#F0E442", "#000000", "#D55E00", "#CC79A7")
@

\section*{Background and Objectives}
\subsection*{Scientific Background}
The budding yeast \textit{Saccharomyces cerevisiae} undergo sporulation under specific starvation conditions of low nitrogen and an absence of fermentable carbon. This process is controlled by a specific gene expression program, and constitutes the meiotic program for the yeast \autocite{Neiman:2011kf}. To ensure that the spores are the proper shape and size, the four meiotic products are enclosed in membranes that are synthesized \textit{de novo}, known as Prospore Membranes (PSMs) \autocite{Neiman:1998tu}. These PSMs serve as templates for the deposition of a multi-layered spore wall, which protects the spore and allows it to survive until more favorable growth conditions return.

This process requires certain conserved pathways, including signaling by the Mitotic Exit Network (MEN), which constitutes the yeast Hippo signaling pathway. CDC15, a kinase that acts upstream of the NDR/LATS kinases Dbf2 and Dbf20, is known to be required for proper spore formation \autocites{PabloHernando:2007cp}{Paulissen:2016gj}
\subsection*{The Dataset}

The dataset I plan to analyzed was published in \textit{Science} in 1998 by Chu \textit{et al.}, in a work titled \enquote{The Transcriptional Program of Sporulation in Budding Yeast} \autocite{Chu:1998tm} . The dataset is publicly available at: \begin{center} \underline{\texttt{http://cmgm.stanford.edu/pbrown/sporulation/additional/spospread.txt}} \end{center}
<<dataset_download, eval = TRUE>>=
# sporulation_data <- read.delim(
#  file = "http://cmgm.stanford.edu/pbrown/sporulation/additional/spospread.txt",
#  header = TRUE,
#  sep = "\t", quote = "\"")
setwd("~/R_Projects/Bioinformatics/final")
sporulation_data <- read.delim(
  file = "spospread.txt",
  header = TRUE,
  sep = "\t", quote = "\"")
expression_matrix <- data.matrix(sporulation_data[70:76])
@
The dataset contains expression data for \Sexpr{nrow(sporulation_data)} genes, and is taken as a timecourse throughout the yeast sporulation program. 

\section*{Computational Methods}
To analyze this dataset, the authors constructed heatmaps showing expression of each identified gene throughout sporulation. I plan to use their data to analyze expression patterns in the context of the genes I am interested in, such as SPS1 and CDC15. This will include making more focused heatmaps, showing the detail in expression patterns for the genes I am most interested in, as well as hierarchical clustering to see which genes out of the entire dataset cluster most similarly with these genes.
<<sample_figures, echo = FALSE>>=
whole_cluster <- hclust(dist(sporulation_data[70:76], method = "euclidean"),
                        method = "complete")
plot(whole_cluster,
     lwd = 2, col = "black",
     ylab="Distance", xlab="Genes",
     main = "Clustering of Gene Expression in Sporulation",
     hang = -1, sub = NA, axes = FALSE)
axis(side = 2, at = seq(0, 20, 4), col = "#0072B2", labels = TRUE, lwd = 2)

sps1 <- grep("sps1$", sporulation_data$SGD.Name, ignore.case = TRUE)
sps1_like <- genefinder(expression_matrix, sps1, numResults = 50)[[1]]
heatmap(expression_matrix[c(sps1, sps1_like$indices),],
        main = "Heatmap of Expression of Sps1-like Genes in Sporulation")
@

Obviously there is some work to be done, but the data is amenable to clustering and making heatmaps! 
     
\section*{Discussion}
Understanding the process of yeast sporulation holds several areas of greater biological relevance. Primarily, the process of closing PSMs and forming spores represents the yeast meiotic cytokinetic event. This is a process that happens in all eukaryotic organisms that undergo meiosis, including mammalian cells and especially humans. 

Furthermore, since many of the genes I am interested in are homologous to members of the Hippo signaling pathway, this research has broader relevance even than meiosis. Hippo signaling is involved in sensing of mechanical tension in higher eukaryotes, and may play a role in linking of spindle tension to cytokinesis. \autocite{Hergovich:2017bv}. Many types of cancers have been known to demonstrate abberrant Hippo activity to continue to proliferate despite physical compression \autocite{Harvey:2013jx}, and therefore a better understanding of Hippo activity in yeast can contribute to an understanding of cancer biology as well. 

\printbibliography

\end{document}